src/portfft/common/subgroup_bluestein.hpp

```diff
@@ -0,0 +1,287 @@
+ * sycl::vec<T, 2>
+ * @param store_modifier Global memory pointer containing the store modifier data, assumed aligned to at least
+ * sycl::vec<T, 2>
+ * @param twiddles_loc view of the local memory containing the twiddles
+ * @param conjugate_on_load Whether or not conjugation of the input is to be done before the fft computation
+ * @param conjugate_on_store Whether or not conjugation of the input is to be done after the fft computation
+ * @param scale_applied Whether or not scale factor is applied
+ * @param scale_factor Value of the scaling factor
+ * @param id_of_wi_in_fft Id of the workitem in the FFT
+ * @param factor_sg Number of workitems participating for one transform
+ * @param factor_wi Number of complex elements per workitem for each transform
+ * @param storage storage scheme of complex values in local memory, SPLIT_COMPLEX or INTERLEAVED_COMPLEX
+ * @param wi_working Whether or not the workitem participates in the data transfers
+ * @param local_imag_offset Number of elements in local memory after which the imaginary component of the values is
+ * stored
+ * @param max_num_batches_local_mem Maximum number of transforms that can be stored in local memory
+ * @param fft_idx_in_local Id of the transform in local memory
+ * @param global_data global_data_struct associated with kernel launch
+ */
+template <Idx SubgroupSize, typename T, typename LocTwiddlesView, typename LocView>
+PORTFFT_INLINE void sg_bluestein_batch_interleaved(
+    T* priv, T* priv_scratch, LocView& loc_view, const T* load_modifier, const T* store_modifier,
+    LocTwiddlesView& twiddles_loc, detail::complex_conjugate conjugate_on_load,
+    detail::complex_conjugate conjugate_on_store, detail::apply_scale_factor scale_applied, T scale_factor,
+    Idx id_of_wi_in_fft, Idx factor_sg, Idx factor_wi, complex_storage storage, bool wi_working, Idx local_imag_offset,
+    Idx max_num_batches_local_mem, Idx fft_idx_in_local, detail::global_data_struct<1>& global_data) {
+  global_data.log_message_global(__func__, "computing forward FFT and applying scaling factor for the backward phase");
+  sg_cooley_tukey<SubgroupSize>(
+      priv, priv_scratch, detail::elementwise_multiply::APPLIED, detail::elementwise_multiply::APPLIED,
+      conjugate_on_load, detail::complex_conjugate::NOT_APPLIED, detail::apply_scale_factor::APPLIED, load_modifier,
+      store_modifier, twiddles_loc, static_cast<T>(1. / (static_cast<T>(factor_sg * factor_wi))), 0, id_of_wi_in_fft,
+      factor_sg, factor_wi, global_data);
+
+  // TODO: Currently local memory is being used to load the data back in natural order for the backward phase, as the
+  // result of sg_dft is transposed. However, the ideal way to this is using shuffles. Implement a batched matrix
+  // transpose to transpose a matrix stored in the private memory of workitems of a subgroup using shuffles only. his we
+  // way can even avoid the 2 sg_bluestein functions that we have today
```

Comment:
Which 2 functions?

Reply:
sg_bluestien packed and sg_bluestein batch_interleaved

Reply:
How does that change let us avoid them?